SG14/algorithm_ext.h

```diff
@@ -80,52 +78,80 @@ namespace stdext
 	template<class BidirIt, class UnaryPredicate>
 	BidirIt unstable_remove_if(BidirIt first, BidirIt last, UnaryPredicate p)
 	{
-		while (1) {
-			while ((first != last) && p(*first)) {
+		while (true) {
+			// Find the first instance of "p"...
+			while (true) {
+				if (first == last) {
+					return first;
+				}
+				if (p(*first)) {
+					break;
+				}
 				++first;
 			}
-			if (first == last--) break;
```

Comment:
Looking at [[bidirectional.iterators]](http://eel.is/c++draft/bidirectional.iterators), that doesn't appear to be the case. `end()--` is defined: prefix `--` does not require a dereferenceable iterator, and postfix `--` is defined in terms of prefix `--` semantics.